PORT-9846-update-gitHub-docs-to-add-examples-that-need-WRITE (#1500)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/resource-mapping-examples.md

# Resource mapping examples

:::warning General permissions limitation with gitHub cloud app

> When using the GitHub Cloud app with Port, certain fields and data points may not be accessible due to the lack of WRITE permissions. These limitations affect advanced repository settings, security features (e.g., code scanning and secret scanning status), and other GitHub objects that require elevated permissions to retrieve.

Reviewer: WRITE API permissions

> :::warning General permissions limitation with gitHub cloud app
> When using the GitHub Cloud app with Port, certain fields and data points may not be accessible due to the lack of WRITE permissions. These limitations affect advanced repository settings, security features (e.g., code scanning and secret scanning status), and other GitHub objects that require elevated permissions to retrieve.

Reviewer: ```suggestion
:::warning General permissions limitation with gitHub cloud app
When using the GitHub Cloud app with Port, certain fields and data points may not be accessible due to the lack of `write` API permissions. These limitations affect advanced repository settings, security features (such as code scanning and secret scanning status), and other GitHub objects that require elevated permissions to retrieve data.
```

If you need to ingest these fields, consider one of the following approaches:

> - Use a self-hosted GitHub app with the appropriate permissions.

Reviewer: Add reference/link to our self hosted app

> - Use a [self-hosted](/build-your-software-catalog/sync-data-to-catalog/git/github/self-hosted-installation) GitHub app with the appropriate permissions.

Reviewer: ```suggestion
- Use our [self-hosted](/build-your-software-catalog/sync-data-to-catalog/git/github/self-hosted-installation) GitHub app which gives you options to enable appropriate `write` permissions.

```

<PortRepoDeploymentAndEnvironmentAppConfig/>

## Map repositories, Dependabot Alerts, and Code scan alerts

> In the following example, you will ingest your GitHub repositories and their alerts (Dependabot and Code scan alerts) to Port. You may use the following Port blueprint definitions and `port-app-config.yml`:

Reviewer 2: ```suggestion
The following example shows how to ingest your GitHub repositories and their alerts (Dependabot and Code scan alerts) into Port. You can use the following Port blueprint definitions and `port-app-config.yml`:
```

<PortRepositoryDependabotAlertMappingAppConfig/>

:::info supported alerts
For Code scan alerts, only open alerts on the default branch are supported.
:::

:::warning limitations with gitHub cloud app

> When using the GitHub Cloud app with Port, certain fields required for ingesting advanced security settings or specific repository configurations may not be accessible due to the lack of WRITE permissions. These fields include but are not limited to:

Reviewer: I thought we already mentioned this above. So I will remove it and just leave the self-hosted option tip

- `allow_squash_merge`
- Advanced security status (e.g., whether code scanning or secret scanning is enabled)

If you need to ingest these fields, consider using a self-hosted GitHub app with the appropriate permissions or creating a GitHub workflow to manually gather and ingest this data into Port.
:::

:::tip self-hosted gitHub app option

> For users who need access to the full range of repository fields, including those requiring WRITE permissions, we recommend setting up a self-hosted GitHub app. This allows full customization of permissions, ensuring all necessary data can be ingested into Port.

Reviewer: ```suggestion
For users who need access to the full range of repository fields, including enabling WRITE permissions, we recommend setting up a self-hosted GitHub app. This allows full customization of permissions, ensuring all necessary data can be ingested into Port.
```

<PortRepositoryDependabotAlertMappingAppConfig/>

:::info supported alerts
For Code scan alerts, only open alerts on the default branch are supported.
:::

- `allow_squash_merge`
- Advanced security status (e.g., whether code scanning or secret scanning is enabled)

If you need to ingest these fields, consider using a self-hosted GitHub app with the appropriate permissions or creating a GitHub workflow to manually gather and ingest this data into Port.

> :::

Reviewer 2: seems like a redundant `:::`, right?

Author: yeah
